File: fibonacci_series/Java/com/example/fibonacci/RecursiveFibonacci.java
package com.example.fibonacci;
import java.math.BigInteger;
import java.util.Scanner;

class RecursiveFibonacci  {
    public static BigInteger recursiveFibonacci(int n) {
        if (n == 0) {
            return BigInteger.ZERO;
        } else if (n == 1) {
            return BigInteger.ONE;
        } else {
            return recursiveFibonacci(n - 1).add(recursiveFibonacci(n - 2));
        }
    }

    public static void printFibonacciSeries(int numberOfTerms) {
        for (int i = 0; i < numberOfTerms; i++) {
            System.out.print(recursiveFibonacci(i));
            if (i != numberOfTerms - 1) {
                System.out.print(", ");
            }
        }
        System.out.println(".");
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the number of terms for the Fibonacci series: ");
        int numberOfTerms = scanner.nextInt();
        scanner.close();

        printFibonacciSeries(numberOfTerms);
    }

}

File: fibonacci_series/Kotlin/recursive_fibonacci.kt
import java.math.BigInteger
import java.util.Scanner

fun recursiveFibonacci(n: Int, a: BigInteger, b: BigInteger, count: Int) {
    if (count < n) {
        val aux = a.add(b)
        println(aux)
        recursiveFibonacci(n, b, aux, count + 1)
    }
}

fun main() {
    print("Enter a number of members in series:")
    val input = Scanner(System.`in`)
    val number = input.nextInt()
    
    val a = BigInteger("0")
    val b = BigInteger("1")
    
    println(a)
    println(b)
    
    recursiveFibonacci(number - 2, a, b, 0)
}
